src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py

```diff
@@ -339,41 +339,46 @@ def create_ldm_bert_config(original_config):
     return config
 
 
-def convert_ldm_unet_checkpoint(checkpoint, config, path=None, extract_ema=False, controlnet=False):
+def convert_ldm_unet_checkpoint(
+    checkpoint, config, path=None, extract_ema=False, controlnet=False, skip_extract_state_dict=False
+):
     """
     Takes a state dict and a config, and returns a converted checkpoint.
     """
 
-    # extract state_dict for UNet
-    unet_state_dict = {}
-    keys = list(checkpoint.keys())
-
-    if controlnet:
-        unet_key = "control_model."
+    if skip_extract_state_dict:
+        unet_state_dict = checkpoint
```

Comment:
this allows us to skip the initial key trimming for when the root level checkpoint doesn't need to be trimmed. If we don't manually skip it, the `unet_state_dict` will be empty because none of the keys have the expected prefix